pennylane/wires.py

```diff
@@ -493,6 +493,207 @@ def unique_wires(list_of_wires):
 
         return Wires(tuple(unique), _override=True)
 
+    def union(self, other):
+        """Return the union of the current Wires object and either another Wires object or an
+        iterable that can be interpreted like a Wires object e.g. List.
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
+
+        Returns:
+            Wires: A new Wires object representing the union of the two Wires objects.
+
+        Raises:
+            TypeError: If ``other`` is not an instance of Wires.
+
+        **Example**
```

Comment:
Should we have such examples: `wires1 | wires2`?

Reply:
Doesn't raise an error here anymore since the function was changed to accept all iterables that could be interpreted as a Wires instance. The error is instead raised in `_process`, which I have not changed.

Reply:
Also, I will move the examples using the pipe operator into the `union()` method since the dunder methods dont show up on the documentation.